controllers/common_test.go

```diff
@@ -0,0 +1,169 @@
+				ObjectMeta: metav1.ObjectMeta{Name: "some-vhost", Namespace: "default"},
+				Spec:       topology.VhostSpec{RabbitmqClusterReference: commonRabbitmqClusterRef},
+			},
+			&topology.SchemaReplication{
+				ObjectMeta: metav1.ObjectMeta{Name: "some-vhost", Namespace: "default"},
+				Spec:       topology.SchemaReplicationSpec{RabbitmqClusterReference: commonRabbitmqClusterRef},
+			},
+			&topology.Federation{
+				ObjectMeta: metav1.ObjectMeta{Name: "some-federation", Namespace: "default"},
+				Spec: topology.FederationSpec{RabbitmqClusterReference: commonRabbitmqClusterRef,
+					UriSecret: &corev1.LocalObjectReference{Name: "federation-uri"}},
+			},
+			&topology.Shovel{
+				ObjectMeta: metav1.ObjectMeta{Name: "some-shovel", Namespace: "default"},
+				Spec: topology.ShovelSpec{RabbitmqClusterReference: commonRabbitmqClusterRef,
+					UriSecret: &corev1.LocalObjectReference{Name: "shovel-uri-secret"}},
+			},
+			&topologyV1alpha.SuperStream{
+				ObjectMeta: metav1.ObjectMeta{Name: "some-super-stream", Namespace: "default"},
+				Spec:       topologyV1alpha.SuperStreamSpec{RabbitmqClusterReference: commonRabbitmqClusterRef},
+			},
+		}
+
+		fakeRabbitMQClient.DeclareBindingReturns(commonHttpCreatedResponse, nil)
+		fakeRabbitMQClient.DeleteBindingReturns(commonHttpDeletedResponse, nil)
+		fakeRabbitMQClient.DeclareExchangeReturns(commonHttpCreatedResponse, nil)
+		fakeRabbitMQClient.DeleteExchangeReturns(commonHttpDeletedResponse, nil)
+		fakeRabbitMQClient.UpdatePermissionsInReturns(commonHttpCreatedResponse, nil)
+		fakeRabbitMQClient.ClearPermissionsInReturns(commonHttpDeletedResponse, nil)
+		fakeRabbitMQClient.PutPolicyReturns(commonHttpCreatedResponse, nil)
+		fakeRabbitMQClient.DeletePolicyReturns(commonHttpDeletedResponse, nil)
+		fakeRabbitMQClient.DeclareQueueReturns(commonHttpCreatedResponse, nil)
+		fakeRabbitMQClient.DeleteQueueReturns(commonHttpDeletedResponse, nil)
+		fakeRabbitMQClient.PutUserReturns(commonHttpCreatedResponse, nil)
+		fakeRabbitMQClient.DeleteUserReturns(commonHttpDeletedResponse, nil)
+		fakeRabbitMQClient.PutVhostReturns(commonHttpCreatedResponse, nil)
+		fakeRabbitMQClient.DeleteVhostReturns(commonHttpDeletedResponse, nil)
+		fakeRabbitMQClient.PutGlobalParameterReturns(commonHttpCreatedResponse, nil)
+		fakeRabbitMQClient.DeleteGlobalParameterReturns(commonHttpDeletedResponse, nil)
+		fakeRabbitMQClient.PutFederationUpstreamReturns(commonHttpCreatedResponse, nil)
+		fakeRabbitMQClient.DeleteFederationUpstreamReturns(commonHttpDeletedResponse, nil)
+		fakeRabbitMQClient.DeclareShovelReturns(commonHttpCreatedResponse, nil)
+		fakeRabbitMQClient.DeleteShovelReturns(commonHttpDeletedResponse, nil)
+	})
+
+	It("sets the domain name in the URI to connect to RabbitMQ", func() {
+		for _, controller := range topologyControllers {
+			controller.SetInternalDomainName(".some-domain.com")
+		}
+
+		for i, _ := range topologyControllers {
+			Expect(client.Create(ctx, topologyObjects[i])).To(Succeed())
```

Comment:
This test was a bit flaky. For some reason, sometimes the test would fail by not adding the internal domain name. By separating in two different loop the "setup" and the "tests", the flake seems to go away.